Gems/ROS2/Code/Source/RobotImporter/URDF/UrdfParser.cpp

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "UrdfParser.h"
+
+#include <fstream>
+
+#include <AzCore/Debug/Trace.h>
+#include <AzCore/std/string/string.h>
+
+#include <urdf_model/model.h>
+
+namespace ROS2
+{
+    namespace UrdfParser::Internal
+    {
+        void CheckIfCurrentLocaleHasDotAsADecimalSeparator()
+        {
+            // Due to the fact that URDF parser takes into account the locale information, incompatibility between URDF file locale and
+            // system locale might lead to incorrect URDF parsing. Mainly it affects floating point numbers, and the decimal separator. When
+            // locales are set to system with comma as decimal separator and URDF file is created with dot as decimal separator, URDF parser
+            // will trim the floating point number after comma. For example, if parsing 0.1, URDF parser will parse it as 0.
+            // This might lead to incorrect URDF loading. If the current locale is not a dot (as per standard ROS locale), we warn the user.
+            std::locale currentLocale("");
+            if (std::use_facet<std::numpunct<char>>(currentLocale).decimal_point() != '.')
+            {
+                AZ_Warning(
+                    "UrdfParser", false, "Locale %s might be incompatible with the URDF file content.\n", currentLocale.name().c_str());
+            }
+        }
+
+        class CustomConsoleHandler : public console_bridge::OutputHandler
+        {
+        private:
+            std::stringstream console_ss;
+
+        public:
+            void log(const std::string& text, console_bridge::LogLevel level, const char* filename, int line) override final;
+
+            //! Clears accumulated log
+            void Clear();
+
+            //! Read accumulated log to file
```

Comment:
Does this return the log in a string? or to a file and the filename is returned in the string?

Reply:
Sorry muscle memory

Reply:
Done